rust/protocol/src/state/signed_prekey.rs

```diff
@@ -1,28 +1,63 @@
 //
-// Copyright 2020 Signal Messenger, LLC.
+// Copyright 2020-2021 Signal Messenger, LLC.
 // SPDX-License-Identifier: AGPL-3.0-only
 //
 
 use crate::proto::storage::SignedPreKeyRecordStructure;
 use crate::{KeyPair, PrivateKey, PublicKey, Result};
 use prost::Message;
 
-pub type SignedPreKeyId = u32;
+/// A unique identifier selecting among this client's known signed pre-keys.
+#[derive(Copy, Clone, Debug, Hash, Eq, PartialEq, PartialOrd, Ord)]
+pub struct SignedPreKeyId(pub u32);
+
+impl From<u32> for SignedPreKeyId {
+    fn from(value: u32) -> Self {
+        Self(value)
+    }
+}
+
+impl From<SignedPreKeyId> for u32 {
+    fn from(value: SignedPreKeyId) -> Self {
+        value.0
+    }
+}
+
+/// A timestamp assigned when creating a [SignedPreKeyRecord].
+#[derive(Copy, Clone, Debug, Hash, Eq, PartialEq, PartialOrd, Ord)]
+pub struct SignedPreKeyTimestamp(pub u64);
```

Comment:
Let's hold off on this one. There's a general consistent "timestamp" type throughout all these APIs which is "milliseconds since 1970", and we can do them all together.

Reply:
Ok, done!